kv: mark subsume as isWrite

kvoli opened the pull request:
Now that `Subsume` is required to force flush all replication streams up
to the lease applied index it froze the range at (https://github.com/cockroachdb/cockroach/pull/136321),
`SubsumeRequest` also needs to be marked as a write in order to pass
through the replication code path, which updates the range controller
force flush index and replicates the force flush index.

Part of: https://github.com/cockroachdb/cockroach/issues/132614
Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> to the non-kvnemesis change

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/136649#-:-ODESbgD9_F1v4lCSdzj:b-ho7l0u)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli)



<!-- Sent from Reviewable.io -->

kvoli commented:
Following some discussion in our weekly meeting, we agreed that this change requires some further thought, as merge is rather subtle. I'm going to conduct further research into potential hazards this change has.

kvoli committed:
kv: mark subsume as isWrite

Now that `Subsume` is required to force flush all replication streams up
to the lease applied index it froze the range at (#136321),
`SubsumeRequest` also needs to be marked as a write in order to pass
through the replication code path, which updates the range controller
force flush index and replicates the force flush index.

Part of: #132614
Release note: None

kvoli commented:
I've removed the kvnemesis change for now. I'm going to make sure the failure that cropped up (https://mesolite.cluster.engflow.com/invocations/default/b6237d65-083d-40e1-afa5-a87f92ea6a83) is well understood before making any testing changes there.
